### DISCUSSION TITLE
URL automation test using appium Uiautomator 2 in java

### DISCUSSION THREAD

**Discussion by soorajtk3** [appium/appium#19341] [Q&A]
I have a public URL.By click on the URL the Application will launch in the Application or browser. So how to automate that?
`package sample;

import io.appium.java_client.AppiumDriver;
import java.net.MalformedURLException;
import java.net.URL;
import java.time.Duration;

import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Link {
	  static AppiumDriver driver;
    public static void main(String[] args) {
     

        try {
        	defaultLogo();
        } catch (Exception e) {
            System.out.println(e.getCause());
            System.out.println(e.getMessage());
            e.printStackTrace();
        }
    }

    public static  void defaultLogo() throws MalformedURLException, Exception {
		DesiredCapabilities cap = new DesiredCapabilities();
		  cap.setCapability("deviceName", "Galaxy A10s");
		  cap.setCapability("ubid", "R9ANA0GCY4J");
		  cap.setCapability("platformName", "Android");
		  cap.setCapability("platformVersion", "11");
		  
		  cap.setCapability("appPackage", "com.sample");
		  cap.setCapability("appActivity", "com.sample.Activity");
		URL url = new URL("http://127.0.0.1:4724/wd/hub");
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(30));
		driver = new AppiumDriver(url, cap);
		
	        
		
		
    }

}`

**Comment by KazuCocoa** [comment]
https://discuss.appium.io/ may help to ask such general questions

**Reply by daraHK88** [reply]
sorry but this link is not working. is there any alternative to "discuss.appium.io"?

**Reply by KazuCocoa** [reply]
No, the website is currently having trouble. We're taking a look if we could bring them back to normal right now.

**Reply by KazuCocoa** [reply]
The website should be ok for now

**Reply by daraHK88** [reply]
thank you. Yes I confirm it's working now. I'm fullstack developer if there is anything that I can help you with please let me know
